torch/csrc/jit/passes/onnx/shape_type_inference.cpp

```diff
@@ -2041,6 +2064,12 @@ void ONNXShapeTypeInference(
       GRAPH_DEBUG(
           "ONNX graph after shape inference: ", prettyPrint(*model_proto));
     }
+  } else if (custom_setType) {
```

Comment:
nit: consider make a function/lambda to get custom_setType, to "lazy execute" it only if needed (when getting to this branch)

Reply:
Done